src/Shared/Components/CICDHistory/LogsRenderer.tsx

```diff
@@ -408,10 +440,41 @@ const LogsRenderer = ({
         }
     }, [handleToggleOpenAllStages])
 
+    const handleNextSearchResult = () => {
+        if (searchResults.results.length > 0) {
+            const currentIndex = (searchResults.currentIndex + 1) % searchResults.results.length
+            setSearchResults({
+                ...searchResults,
+                currentIndex,
+            })
+            setStageList(getStageListFromStreamData(currentIndex, searchKey))
+        }
+    }
+
+    const handlePrevSearchResult = () => {
+        if (searchResults.results.length > 0) {
+            const currentIndex =
+                searchResults.currentIndex > 0 ? searchResults.currentIndex - 1 : searchResults.results.length - 1
+            setSearchResults({
+                ...searchResults,
+                currentIndex,
+            })
+            setStageList(getStageListFromStreamData(currentIndex, searchKey))
+        }
+    }
+
     const handleSearchEnter = (searchText: string) => {
         handleSearch(searchText)
-        const newStageList = getStageListFromStreamData(searchText)
-        setStageList(newStageList)
+        if (searchKey === searchText) {
+            handleNextSearchResult()
+        } else {
+            const currentIndex = 0
+            setSearchResults((prev) => ({
```

Comment:
Can create a method that recieves currentIndex and abstract out setSearchResults, setStageList(getStageListFromStreamData(currentIndex, searchText)) for handleSearchEnter, handlePrevSearchResult, handleNextSearchResult

Reply:
abstracted that whole chunk including currentIndex calculation into a method that accepts type of cycle - 'prev' or 'next' or 'reset'